### PR TITLE
Add Spacy Japanese pre-trained model into Rasa Docker image (#7492)

### PR DESCRIPTION
**Pull request by jnoelvictorino**
**Proposed changes**:
* Update `spacy` to `2.3.2` in `pyproject.toml`
* Update poetry.lock during build
* Include `./docker/Dockerfile_pretrained_embeddings_spacy_en_ja` to add `ja_core_news_md`

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [X] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)

### CODE REVIEW

**Comment by sanchariGr** [comment]
Hi @jnoelvictorino since its been a while this PR was created, could you please rebase with master?

File: changelog/7492.improvement.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Added `./docker/Dockerfile_pretrained_embeddings_spacy_en_ja` to include Spacy's Japanese pre-trained model `ja_core_news_md`.
+
+Updated `spacy` to `2.3.2` and `poetry.lock` during the process of adding Japanese pre-trained model.
```
Comment (ancalita) [inline changelog/7492.improvement.md:3]: This won't be necessary after you've updated your branch with the latest changes from `main` as spacy is already upgraded to version `3.2`.

File: docker/Dockerfile_pretrained_embeddings_spacy_en_ja [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  poetry build -f wheel -n && \
+  pip install --no-deps dist/*.whl && \
+  rm -rf dist *.egg-info
+
+# make sure we use the virtualenv
+ENV PATH="/opt/venv/bin:$PATH"
+
+# spacy link
+RUN python -m spacy download en_core_web_md && \
+    python -m spacy link en_core_web_md en && \
```
Comment (ancalita) [inline docker/Dockerfile_pretrained_embeddings_spacy_en_ja:54]: Why is this download required if you're building an image with the Japanese pretrained language model?

File: docker/Dockerfile_pretrained_embeddings_spacy_en_ja [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+# install poetry
+# keep this in sync with the version in pyproject.toml and Dockerfile
+ENV POETRY_VERSION 1.0.5
+RUN curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python
+ENV PATH "/root/.poetry/bin:/opt/venv/bin:${PATH}"
+
+# copy files
+COPY . /build/
+COPY docker/configs/config_pretrained_embeddings_spacy_en.yml /build/config.yml
```
Comment (ancalita) [inline docker/Dockerfile_pretrained_embeddings_spacy_en_ja:35]: I think you will need to create a different config yml file using the Japanese language model.

**Review by ancalita — COMMENTED** [review]
@jnoelvictorino Thank you for your contribution!

Could you please follow the naming convention we have in our `docker` folder and label this file `Dockerfile.pretrained_embeddings_spacy_en_ja`. In addition, please follow the format used in existing Dockerfiles, you could look at `Dockerfile.pretrained_embeddings_spacy_de` for example.
